pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -265,21 +291,27 @@ func (o *EditOptions) Run() error {
 			}
 			// not a syntax error as it turns out...
 			containsError = false
-			updatedVisitor := resource.InfoListVisitor(updatedInfos)
 
-			// need to make sure the original namespace wasn't changed while editing
-			if err := updatedVisitor.Visit(resource.RequireNamespace(o.CmdNamespace)); err != nil {
-				return preservedFile(err, file, o.ErrOut)
-			}
+			var updatedVisitor resource.Visitor
+			if o.EditMode != ApplyEditMode {
```

Comment:
can we keep the branching based on the inside visitAnnotation? that is already checking `o.ApplyAnnotation`